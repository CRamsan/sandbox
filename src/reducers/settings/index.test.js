import reducer from '.'

const initialState = {
  xMin: -1,
  yMin: -1,
  zMin: -4,
  xMax: 1,
  yMax: 1,
  zMax: 4,
  segments: 30,
  functionColor: '#bada55',
}

describe('settings', () => {
  it('should return the initial state', () => {
    const next = reducer(initialState, {})
    expect(next).toEqual(initialState)
  })

  it('SETTINGS_SET_X_MIN', () => {
    const next = reducer(initialState, { type: 'SETTINGS_SET_X_MIN', xMin: -999 })
    expect(next).toEqual({ ...initialState, xMin: -999 })
    expect(next).toMatchSnapshot()
  })

  it('SETTINGS_SET_Y_MIN', () => {
    const next = reducer(initialState, { type: 'SETTINGS_SET_Y_MIN', yMin: -999 })
    expect(next).toEqual({ ...initialState, yMin: -999 })
    expect(next).toMatchSnapshot()
  })

  it('SETTINGS_SET_Z_MIN', () => {
    const next = reducer(initialState, { type: 'SETTINGS_SET_Z_MIN', zMin: -999 })
    expect(next).toEqual({ ...initialState, zMin: -999 })
    expect(next).toMatchSnapshot()
  })

  it('SETTINGS_SET_X_MAX', () => {
    const next = reducer(initialState, { type: 'SETTINGS_SET_X_MAX', xMax: 999 })
    expect(next).toEqual({ ...initialState, xMax: 999 })
    expect(next).toMatchSnapshot()
  })

  it('SETTINGS_SET_Y_MAX', () => {
    const next = reducer(initialState, { type: 'SETTINGS_SET_Y_MAX', yMax: 999 })
    expect(next).toEqual({ ...initialState, yMax: 999 })
    expect(next).toMatchSnapshot()
  })

  it('SETTINGS_SET_Z_MAX', () => {
    const next = reducer(initialState, { type: 'SETTINGS_SET_Z_MAX', zMax: 999 })
    expect(next).toEqual({ ...initialState, zMax: 999 })
    expect(next).toMatchSnapshot()
  })

  it('SETTINGS_SET_SEGMENTS', () => {
    const next = reducer(initialState, { type: 'SETTINGS_SET_SEGMENTS', segments: 15 })
    expect(next).toEqual({ ...initialState, segments: 15 })
    expect(next).toMatchSnapshot()
  })

  it('SETTINGS_SET_FUNCTION_COLOR', () => {
    const next = reducer(initialState, { type: 'SETTINGS_SET_FUNCTION_COLOR', functionColor: '#fff' })
    expect(next).toEqual({ ...initialState, functionColor: '#fff' })
    expect(next).toMatchSnapshot()
  })
})
