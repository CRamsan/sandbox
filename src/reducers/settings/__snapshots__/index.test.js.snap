// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings SETTINGS_SET_FUNCTION_COLOR 1`] = `
Object {
  "functionColor": "#fff",
  "segments": 30,
  "xMax": 1,
  "xMin": -1,
  "yMax": 1,
  "yMin": -1,
  "zMax": 4,
  "zMin": -4,
}
`;

exports[`settings SETTINGS_SET_SEGMENTS 1`] = `
Object {
  "functionColor": "#bada55",
  "segments": 15,
  "xMax": 1,
  "xMin": -1,
  "yMax": 1,
  "yMin": -1,
  "zMax": 4,
  "zMin": -4,
}
`;

exports[`settings SETTINGS_SET_X_MAX 1`] = `
Object {
  "functionColor": "#bada55",
  "segments": 30,
  "xMax": 999,
  "xMin": -1,
  "yMax": 1,
  "yMin": -1,
  "zMax": 4,
  "zMin": -4,
}
`;

exports[`settings SETTINGS_SET_X_MIN 1`] = `
Object {
  "functionColor": "#bada55",
  "segments": 30,
  "xMax": 1,
  "xMin": -999,
  "yMax": 1,
  "yMin": -1,
  "zMax": 4,
  "zMin": -4,
}
`;

exports[`settings SETTINGS_SET_Y_MAX 1`] = `
Object {
  "functionColor": "#bada55",
  "segments": 30,
  "xMax": 1,
  "xMin": -1,
  "yMax": 999,
  "yMin": -1,
  "zMax": 4,
  "zMin": -4,
}
`;

exports[`settings SETTINGS_SET_Y_MIN 1`] = `
Object {
  "functionColor": "#bada55",
  "segments": 30,
  "xMax": 1,
  "xMin": -1,
  "yMax": 1,
  "yMin": -999,
  "zMax": 4,
  "zMin": -4,
}
`;

exports[`settings SETTINGS_SET_Z_MAX 1`] = `
Object {
  "functionColor": "#bada55",
  "segments": 30,
  "xMax": 1,
  "xMin": -1,
  "yMax": 1,
  "yMin": -1,
  "zMax": 999,
  "zMin": -4,
}
`;

exports[`settings SETTINGS_SET_Z_MIN 1`] = `
Object {
  "functionColor": "#bada55",
  "segments": 30,
  "xMax": 1,
  "xMin": -1,
  "yMax": 1,
  "yMin": -1,
  "zMax": 4,
  "zMin": -999,
}
`;
